File: integration-tests/iac/lambdas/python/sns_producer.py
import json
import os
import boto3

sns = boto3.client('sns')


def handler(event, context):
    topic_arn = os.environ['TOPIC_ARN']

    message = {
        'message': 'Hello from SNS producer!',
        'request_id': context.aws_request_id,
    }

    response = sns.publish(
        TopicArn=topic_arn,
        Message=json.dumps(message),
        Subject='Test Message',
    )

    print(f"Published message to SNS: {response['MessageId']}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'message_id': response['MessageId'],
        }),
    }

File: integration-tests/iac/lambdas/node/sns_producer.mjs
import { SNSClient, PublishCommand } from '@aws-sdk/client-sns';

const sns = new SNSClient();

export async function handler(event, context) {
    const topicArn = process.env.TOPIC_ARN;

    const message = {
        message: 'Hello from SNS producer!',
        request_id: context.awsRequestId,
    };

    const response = await sns.send(new PublishCommand({
        TopicArn: topicArn,
        Message: JSON.stringify(message),
        Subject: 'Test Message',
    }));

    console.log(`Published message to SNS: ${response.MessageId}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            message_id: response.MessageId,
        }),
    };
}
